File: src/main/java/org/example/bankramenserver/domain/transaction/presentation/dto/RecentTransactionListResponse.java
package org.example.bankramenserver.domain.transaction.presentation.dto;

import io.swagger.v3.oas.annotations.media.Schema;
import lombok.Builder;

import java.util.List;

@Builder
@Schema(description = "최근 거래 내역 목록 응답")
public record RecentTransactionListResponse(
        @Schema(description = "최근 거래 내역 목록")
        List<TransactionHistoryResponse> transactions
) {

    public static RecentTransactionListResponse from(List<TransactionHistoryResponse> transactions) {
        return RecentTransactionListResponse.builder()
                .transactions(transactions)
                .build();
    }
}

File: src/main/kotlin/org/example/bankramenserver/domain/transaction/presentation/dto/RecentTransactionListResponse.kt
package org.example.bankramenserver.domain.transaction.presentation.dto

import io.swagger.v3.oas.annotations.media.Schema

@Schema(description = "최근 거래 내역 목록 응답")
data class RecentTransactionListResponse(
    @field:Schema(description = "최근 거래 내역 목록") val transactions: List<TransactionHistoryResponse>,
) {
    fun transactions() = transactions

    companion object {
        @JvmStatic fun from(transactions: List<TransactionHistoryResponse>) = RecentTransactionListResponse(transactions)
    }
}
